app/models/team.rb

```diff
@@ -60,10 +70,39 @@ def self.policy_class
     TeamPolicy
   end
 
+  def password_bytesize
+    Crypto::Symmetric::ALGORITHMS[encryption_algorithm].password_bitsize
+  end
+
+  def encryption_algorithm
+    # using read_attribute to avoid recursion
+    read_attribute(:encryption_algorithm) || Crypto::Symmetric.latest_algorithm
+  end
+
+  def encryption_algorithm=(algorithm)
+    return if Crypto::Symmetric.all.exclude?(algorithm)
+
+    write_attribute(:encryption_algorithm, algorithm)
+  end
+
+  def new_team_password
+    Crypto::Symmetric::ALGORITHMS[encryption_algorithm].random_key
+  end
+
+  def update_encryption_algorithm
```

Comment:
set this from outside in recrypt domain class
team model should not care about this